lib/one_model.ex

```diff
@@ -154,60 +154,98 @@ defmodule OneModel do
       @doc """
       Get a list of #{@schema}'s.
 
-      ## Options'
+      ## Options
 
       * `preload: list`
+      * `order_by: atom | list`
+      * `select: atom | list`
+        * Passing an atom will return the value of the given atom key 
+        * Passing a list of keys will return a map with the {key, value} pairs
+      * `limit: integer`
       """
       @spec list(keyword()) :: [t()]
       def list(opts \\ []) do
-        if preload = opts[:preload] do
-          @schema
-          |> preload(^preload)
-          |> order_by(asc: :inserted_at)
-          |> @repo.all
-        else
-          @repo.all(@schema)
-        end
+        @schema
+        |> do_order(opts[:order_by])
+        |> do_preload(opts[:preload])
+        |> do_select(opts[:select])
+        |> do_limit(opts[:limit])
+        |> @repo.all()
       end
 
       @doc """
-      Get a list of #{@schema},s given a list of {field, value} pairs.
+      Get a list of #{@schema}s given a list of {field, value} pairs.
 
       ## Preload
 
       Pass a list of preloads with the `:preload` key.
 
+      ## Order by
+
+      Pass an atom or list of fields to sort by with the `:order_by` key.
+      Defaults to sorting by `:inserted_at` in ascending order.
+
+      ## Select
+
+      Pass an atom or list of fields to be selected with the `:select` key.
+      Passing an atom will return the value of the given atom key.
+      Passing a list of keys will return a map with the {key, value} pairs.
+
+      ## Limit
+
+      Pass an integer with the `:limit` key to specify the number of entries returned.
+
       ## Examples
 
-          #{@schema}.list_by(field1: value1, field2: field2, preload: [:association])
+          #{@schema}.list_by(field1: value1, field2: value2, preload: [:association])
+          #{@schema}.list_by(field1: value1, order_by: :field2)
+          #{@schema}.list_by(field1: value1, order_by: [asc: :field3])
+          #{@schema}.list_by(field1: value1, select: :field2)
+          #{@schema}.list_by(field2: value2, select: [:field1, field3], limit: 2)
       """
       @spec list_by(keyword()) :: [t()]
       def list_by(opts) do
-        opts
+        (opts ++ [order_by: :inserted_at])
```

Comment:
It's not obvious that by appending the order_by tuple, we are essentionally not overriding an existing order_by. A more expressive solution would be 
```
opts 
|> Keyword.put_new(:order_by, :inserted_at)
|> ...
```